x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.planner;
+
+import org.elasticsearch.xpack.ql.QlIllegalArgumentException;
+
+public class QueryFolderFailTests extends AbstractQueryFolderTestCase {
+    public void testPropertyEquationFilterUnsupported() {
+        QlIllegalArgumentException e = expectThrows(QlIllegalArgumentException.class,
+                () -> plan("process where (serial_event_id<9 and serial_event_id >= 7) or (opcode == pid)"));
+        String msg = e.getMessage();
+        assertEquals("Line 1:74: Comparisons against variables are not (currently) supported; offender [pid] in [==]", msg);
+    }
+
+    public void testPropertyEquationInClauseFilterUnsupported() {
+        QlIllegalArgumentException e = expectThrows(QlIllegalArgumentException.class,
+                () -> plan("process where opcode in (1,3) and process_name in (parent_process_name, \"SYSTEM\")"));
+        String msg = e.getMessage();
+        assertEquals("Line 1:52: Comparisons against variables are not (currently) supported; offender [parent_process_name] in [==]", msg);
```

Comment:
This error message is confusing to the end user: there is no equality comparison in the query... I'm not saying the test is incorrect, but the way we report the error in the context of EQL is slightly incorrect.

Reply:
Agreed. @costin thoughts?

Reply:
I agree - to improve the message one could either look and check whether it's the first comparison (which might not work if `any` is added) or check whether the comparison is added against the event_type field and if so, mention that in the error message `(e.g. offender `eventy.category`).

Reply:
Will open a separate issue for the error message per conversation with Costin.